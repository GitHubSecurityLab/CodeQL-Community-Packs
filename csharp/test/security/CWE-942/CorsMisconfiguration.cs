using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Mvc;
using System;
using Microsoft.Extensions.DependencyInjection;

public class Test {
  public void ConfigureServices(string[] args) {
    var builder = WebApplication.CreateBuilder(args);
    var MyAllowSpecificOrigins = "_myAllowSpecificOrigins";

    builder.Services.AddCors(options => {
      options.AddPolicy(MyAllowSpecificOrigins,
        policy => {
          policy.AllowAnyOrigin().AllowAnyHeader().AllowAnyMethod();
        });
      options.AddDefaultPolicy(
        builder => builder
        .WithOrigins(["*"])
        .AllowAnyMethod()
        .AllowAnyHeader());
    });

    var app = builder.Build();

    app.MapGet("/", () => "Hello World!");
    app.UseCors(MyAllowSpecificOrigins);

    app.Run();
  }
}