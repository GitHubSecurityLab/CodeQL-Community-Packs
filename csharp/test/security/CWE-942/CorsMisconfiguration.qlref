security/CWE-942/CorsMisconfiguration.ql