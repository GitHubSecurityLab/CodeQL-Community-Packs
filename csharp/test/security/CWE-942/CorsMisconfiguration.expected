| CorsMisconfiguration.cs:12:7:15:10 | call to method AddPolicy | The following CORS policy may allow requests from 3rd party websites |
| CorsMisconfiguration.cs:16:7:20:26 | call to method AddDefaultPolicy | The following CORS policy may allow requests from 3rd party websites |
